bundle/config/workspace.go

```diff
@@ -114,14 +114,23 @@ func (w *Workspace) Client() (*databricks.WorkspaceClient, error) {
 		}
 	}
 
-	if w.Profile != "" && w.Host != "" {
+	// Resolve the configuration. This is done by [databricks.NewWorkspaceClient] as well, but here
+	// we need to verify that a profile, if loaded, matches the host configured in the bundle.
+	err := cfg.EnsureResolved()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
After this, all environment variables, and optionally a profile, have been loaded.